static/app/views/dashboardsV2/widgetCard/widgetCardContextMenu.tsx

```diff
@@ -119,36 +138,31 @@ function WidgetCardContextMenu({
         ...discoverLocation.query,
       })}`;
       if (widget.queries.length === 1) {
-        menuOptions.push(
-          <Link
-            key="open-discover-link"
-            to={discoverPath}
-            onClick={() => {
-              trackAdvancedAnalyticsEvent('dashboards_views.open_in_discover.opened', {
-                organization,
-                widget_type: widget.displayType,
-              });
-            }}
-          >
-            <StyledMenuItem key="open-discover">{t('Open in Discover')}</StyledMenuItem>
-          </Link>
-        );
+        menuOptions.push({
```

Comment:
both are too similar...what if we write something like:
```

 menuOptions.push({
        key: 'open-discover',
        label: t('Open in Discover'),
        leadingItems: <IconTelescope />,
        to: widget.queries.length === 1 ? discoverPath : undefined,
        onAction: () => {
          trackAdvancedAnalyticsEvent('dashboards_views.query_selector.opened', {
            organization,
            widget_type: widget.displayType,
          });

          if (widget.queries.length === 1) {
            openDashboardWidgetQuerySelectorModal({organization, widget});
          }
        },
      });
```

Reply:
oh this is much better!